### ISSUE TITLE
Installation fails due to repository not reachable

### ISSUE THREAD

**Issue by mondsen** [diglos/pi-gen#5]
Hi,
I am trying to install on RPi4 8GB with 1TB SSD, but installation fails. No services are started. I checked the syslog and it seems like your repo apt.ethraspbian.com can not be reached so no software is installed and the node can not start.

What I see from the logs is this:

> Jul  8 16:14:23 ubuntu dbus-daemon[1262]: [system] Successfully activated service 'org.freedesktop.PackageKit'
Jul  8 16:14:23 ubuntu systemd[1]: Started PackageKit Daemon.
Jul  8 16:14:48 ubuntu rc.local[3041]: Fetched 10.7 MB in 26s (408 kB/s)
Jul  8 16:14:50 ubuntu rc.local[3041]: Reading package lists...
Jul  8 16:14:50 ubuntu rc.local[3041]: W: Failed to fetch http://apt.ethraspbian.com/dists/bionic/InRelease  Temporary failure resolving 'apt.ethraspbian.com'
Jul  8 16:14:50 ubuntu rc.local[3041]: W: Some index files failed to download. They have been ignored, or old ones used instead.
Jul  8 16:14:50 ubuntu rc.local[1295]: Installing Ethereum packages
Jul  8 16:14:50 ubuntu rc.local[3513]: Reading package lists...
Jul  8 16:14:50 ubuntu rc.local[3513]: Building dependency tree...
Jul  8 16:14:50 ubuntu rc.local[3513]: Reading state information...
Jul  8 16:14:51 ubuntu rc.local[3513]: The following NEW packages will be installed:
Jul  8 16:14:51 ubuntu rc.local[3513]:   geth
Jul  8 16:14:56 ubuntu systemd-resolved[1176]: Using degraded feature set (TCP) for DNS server fd00:4697:3d81:b810:c8e5::.
Jul  8 16:15:05 ubuntu systemd-resolved[1176]: Using degraded feature set (UDP) for DNS server fd00:4697:3d81:b810:c8e5::.
Jul  8 16:15:08 ubuntu systemd-resolved[1176]: Using degraded feature set (TCP) for DNS server fd00:4697:3d81:b810:c8e5::.
Jul  8 16:15:14 ubuntu systemd-resolved[1176]: Using degraded feature set (UDP) for DNS server fd00:4697:3d81:b810:c8e5::.
Jul  8 16:15:17 ubuntu systemd-resolved[1176]: Using degraded feature set (TCP) for DNS server fd00:4697:3d81:b810:c8e5::.
Jul  8 16:15:19 ubuntu rc.local[3513]: 0 upgraded, 1 newly installed, 0 to remove and 89 not upgraded.
Jul  8 16:15:19 ubuntu rc.local[3513]: Need to get 13.8 MB of archives.
Jul  8 16:15:19 ubuntu rc.local[3513]: After this operation, 0 B of additional disk space will be used.
Jul  8 16:15:19 ubuntu rc.local[3513]: Err:1 http://apt.ethraspbian.com bionic-security/main arm64 geth arm64 1.9.5-0
Jul  8 16:15:19 ubuntu rc.local[3513]:   Temporary failure resolving 'apt.ethraspbian.com'
Jul  8 16:15:19 ubuntu rc.local[3513]: E: Failed to fetch http://apt.ethraspbian.com/pool/main/g/geth/geth_1.9.5-0_arm64.deb  Temporary failure resolving 'apt.ethraspbian.com'
Jul  8 16:15:19 ubuntu rc.local[3513]: E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
Jul  8 16:15:19 ubuntu rc.local[3536]: Failed to stop geth.service: Unit geth.service not loaded.
Jul  8 16:15:19 ubuntu rc.local[3537]: Reading package lists...
Jul  8 16:15:19 ubuntu rc.local[3537]: Building dependency tree...
Jul  8 16:15:19 ubuntu rc.local[3537]: Reading state information...
Jul  8 16:15:19 ubuntu rc.local[3537]: Package swarm is not available, but is referred to by another package.
Jul  8 16:15:19 ubuntu rc.local[3537]: This may mean that the package is missing, has been obsoleted, or
Jul  8 16:15:19 ubuntu rc.local[3537]: is only available from another source
Jul  8 16:15:20 ubuntu rc.local[3537]: E: Unable to locate package nethermind
Jul  8 16:15:20 ubuntu rc.local[3537]: E: Unable to locate package hyperledger-besu
Jul  8 16:15:20 ubuntu rc.local[3537]: E: Package 'swarm' has no installation candidate
Jul  8 16:15:20 ubuntu rc.local[3537]: E: Unable to locate package ipfs
Jul  8 16:15:20 ubuntu rc.local[3537]: E: Unable to locate package raiden
Jul  8 16:15:20 ubuntu rc.local[3537]: E: Unable to locate package statusd
Jul  8 16:15:20 ubuntu rc.local[3537]: E: Unable to locate package vipnode
Jul  8 16:15:20 ubuntu rc.local[3537]: E: Unable to locate package prysm-beacon
Jul  8 16:15:20 ubuntu rc.local[3537]: E: Unable to locate package prysm-validator
Jul  8 16:15:20 ubuntu rc.local[3537]: E: Unable to locate package prysm-slasher
Jul  8 16:15:20 ubuntu rc.local[3537]: E: Unable to locate package lighthouse
Jul  8 16:15:20 ubuntu rc.local[3550]: --2020-07-08 16:15:20--  https://dl.grafana.com/oss/release/grafana_6.7.3_arm64.deb
Jul  8 16:15:23 ubuntu systemd-resolved[1176]: Using degraded feature set (UDP) for DNS server fd00:4697:3d81:b810:c8e5::.
Jul  8 16:15:29 ubuntu systemd-resolved[1176]: Using degraded feature set (TCP) for DNS server fd00:4697:3d81:b810:c8e5::.
Jul  8 16:15:35 ubuntu systemd-resolved[1176]: Using degraded feature set (UDP) for DNS server fd00:4697:3d81:b810:c8e5::.
Jul  8 16:15:36 ubuntu rc.local[3550]: Resolving dl.grafana.com (dl.grafana.com)... failed: Temporary failure in name resolution.
Jul  8 16:15:36 ubuntu rc.local[3550]: wget: unable to resolve host address ‘dl.grafana.com’
Jul  8 16:15:36 ubuntu rc.local[3551]: dpkg: error: cannot access archive '/tmp/grafana_6.7.3_arm64.deb': No such file or directory
Jul  8 16:15:36 ubuntu rc.local[3563]: WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

Any idea what it wrong or what I can do?

When I perform apt update now apt.ethraspbian.com can be reached. Very strange to me. Can this be related?

`Jul  8 16:13:50 ubuntu rc.local[2831]: Executing: /tmp/apt-key-gpghome.Aur2l5jpg9/gpg.1.sh --keyserver keyserver.ubuntu.com --recv-keys 8A584409D327B0A5
Jul  8 16:13:50 ubuntu systemd-resolved[1176]: Server returned error NXDOMAIN, mitigating potential DNS violation DVE-2018-0001, retrying transaction with reduced feature level UDP.
Jul  8 16:13:50 ubuntu systemd-resolved[1176]: message repeated 2 times: [ Server returned error NXDOMAIN, mitigating potential DNS violation DVE-2018-0001, retrying transaction with reduced feature level UDP.]
Jul  8 16:13:51 ubuntu rc.local[2890]: gpg: key 96B52AFDF8C1BD5E: public key "Diego Losada Eth On ARM <diego.losadaf@gmail.com>" imported
Jul  8 16:13:51 ubuntu rc.local[2890]: gpg: Total number processed: 1
Jul  8 16:13:51 ubuntu rc.local[2890]: gpg:               imported: 1`

**Comment by mondsen** [comment]
**Update**
I managed to get everything running. What I did: after I figured out the scripts did not run successful due to connection issues to repo I monitored an unattended upgrade in the background. I also saw that _apt update_ was able to retrieve information from the ethraspbian repo. So I deleted the first-run.flag file and started everything over. 

This time all files were downloaded and installed. I was able to oben Grafana, but Prometheus was not running. So I installed Prometheus again and now everything seems to work. 

TL;DR: maybe we need an FAQ that answers the question on how to start over (remove first-run.flag). And for unknown reasons there can be connection issues regarding the repo.

**Comment by glamperd** [comment]
I saw similar symptoms to these when I first started - notably the errors when doing `sudo apt update`. In my case that improved after I did this: `sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 8A584409D327B0A5`.
`sudo apt update` then worked OK. Once geth was installed it started syncing. Not sure that's the case for other packages. Where is this 'first-run.flag' located?

**Comment by mondsen** [comment]
FLAG="/root/first-run.flag"

**Comment by glamperd** [comment]
Yes, thanks. I found it, removed it, and rebooted. Looks good now.

**Comment by yan-zhongbao** [comment]
> **Update**
> I managed to get everything running. What I did: after I figured out the scripts did not run successful due to connection issues to repo I monitored an unattended upgrade in the background. I also saw that _apt update_ was able to retrieve information from the ethraspbian repo. So I deleted the first-run.flag file and started everything over.
> 
> This time all files were downloaded and installed. I was able to oben Grafana, but Prometheus was not running. So I installed Prometheus again and now everything seems to work.
> 
> TL;DR: maybe we need an FAQ that answers the question on how to start over (remove first-run.flag). And for unknown reasons there can be connection issues regarding the repo.

hello, as you said, I installed Prometheus by myself, and then I start Grafana. but when I browsed  Grafana , the password was not  'ethereum' , and there's no dashboard when I logged in. Do you know how to fix this ? thanks.

**Comment by diglos** [comment]
Hi, this is usually related to a failed connection to the repo. The easy way of solving this is by flashing again the image.
